modules/service/src/main/scala/lucuma/odb/service/GuideEnvironmentService.scala

```diff
@@ -0,0 +1,575 @@
+import lucuma.core.enums.Site
+import lucuma.core.geom.ShapeExpression
+import lucuma.core.geom.gmos.probeArm
+import lucuma.core.geom.jts.interpreter.given
+import lucuma.core.math.Angle
+import lucuma.core.math.Coordinates
+import lucuma.core.math.Offset
+import lucuma.core.math.ProperMotion
+import lucuma.core.math.Wavelength
+import lucuma.core.model.CatalogInfo
+import lucuma.core.model.ConstraintSet
+import lucuma.core.model.ElevationRange
+import lucuma.core.model.ElevationRange.AirMass
+import lucuma.core.model.ElevationRange.HourAngle
+import lucuma.core.model.EphemerisKey
+import lucuma.core.model.ObjectTracking
+import lucuma.core.model.Observation
+import lucuma.core.model.PosAngleConstraint
+import lucuma.core.model.Program
+import lucuma.core.model.SiderealTracking
+import lucuma.core.model.SourceProfile
+import lucuma.core.model.Target
+import lucuma.core.model.Target.Sidereal
+import lucuma.core.model.User
+import lucuma.core.model.sequence.ExecutionConfig
+import lucuma.core.model.sequence.InstrumentExecutionConfig.GmosNorth
+import lucuma.core.model.sequence.InstrumentExecutionConfig.GmosSouth
+import lucuma.core.model.sequence.Step
+import lucuma.core.model.sequence.StepConfig
+import lucuma.itc.client.ItcClient
+import lucuma.odb.data.PosAngleConstraintMode
+import lucuma.odb.data.TargetRole
+import lucuma.odb.json.all.query.given
+import lucuma.odb.logic.Generator
+import lucuma.odb.logic.Generator.FutureLimit
+import lucuma.odb.logic.PlannedTimeCalculator
+import lucuma.odb.sequence.util.CommitHash
+import lucuma.odb.util.Codecs.*
+import lucuma.odb.util.GmosCodecs.*
+import org.http4s.Header
+import org.http4s.Headers
+import org.http4s.Method
+import org.http4s.Request
+import org.http4s.client.Client
+import skunk.AppliedFragment
+import skunk.Decoder
+import skunk.circe.codec.all.*
+import skunk.codec.all.*
+import skunk.implicits.*
+
+import java.time.Instant
+import java.time.LocalDateTime
+import java.time.Month
+import java.time.ZoneOffset
+import java.time.temporal.ChronoField
+import java.time.temporal.ChronoUnit
+
+import Services.Syntax.*
+
+trait GuideEnvironmentService[F[_]] {
+  import GuideEnvironmentService.Error
+  import GuideEnvironmentService.GuideEnvironment
+
+  def get(pid: Program.Id, oid: Observation.Id, obsTime: Instant)(using
+    NoTransaction[F]
+  ): F[Either[Error, GuideEnvironment]]
+}
+
+object GuideEnvironmentService {
+  // There is only one now, will eventually need to get this from somewhere.
+  val guideProbe: GuideProbe = GuideProbe.GmosOiwfs
+
+  case class GuideTarget(probe: GuideProbe, target: Sidereal)
+
+  object GuideTarget {
+    given Encoder[GuideTarget] =
+      Encoder.instance { gt =>
+        Json.obj(
+          "probe"         -> gt.probe.asJson,
+          "name"          -> gt.target.name.asJson,
+          "sourceProfile" -> gt.target.sourceProfile.asJson,
+          "sidereal"      -> gt.target.asJson
+        )
+      }
+  }
```

Comment:
Oh, yeah. I remember one reason I had for not making the `GuideTarget` more "publicly" available. Clients can simply use the `Target` model decoder for the GuideTarget because the Target model does not include the `Target.Id` or the `Program`.  Explore decodes the API result into a `(Target.Id, Target)`. For this, it could just decode into a `(GuideProbe, Target)`. I was just thinking of `GuideTarget` as a convenience class for this service.